saltwater-repl/main.rs

```diff
@@ -0,0 +1,17 @@
+//! The repl implementation for saltwater.
+#![deny(rust_2018_idioms)]
+
+mod commands;
+mod helper;
+mod repl;
+
+fn main() {
+    let mut repl = repl::Repl::new();
+    match repl.run() {
+        Ok(_) => {}
+        Err(err) => {
+            println!("error: {}", err);
```

Comment:
Hmm, these errors are only sometimes showing up.
```
>> "a";
error: expression returns unsupported type: Array(Char(true), Fixed(2))
>> @
# I pressed ctrl+c
>> x = 1

>>
```

Reply:
These lines only print the error that is returned by `rustyline` if something terminal related fails. 
The input isn't valid, because the `Validator` checks if the input is a valid expression.

Reply:
Right, but this isn't reporting the error at _all_, which is a bug. Do you know how to fix that? I'm not sure entirely where the syntax errors are bubbling up - maybe in `execute_code`?

Reply:
The errors are reported in `process_line`